qlty-cli/src/commands/check.rs

```diff
@@ -307,6 +308,16 @@ impl Check {
                 bail!("Missing remote commit ID from Git pre-push input")
             }
 
+            // Check if the remote commit ID is present locally
+            let remote_commit_present_locally = cmd!("git", "cat-file", "-e", remote_commit_id)
```

Comment:
I wonder if this can be done with the `git2` crate instead of needing to shell out to a `git` sub-command